src/psyclone/domain/lfric/transformations/lfric_loop_fuse_trans.py

```diff
@@ -120,15 +120,24 @@ def validate(self, node1, node2, options=None):
         # Call the parent class validation first
 
         if not options:
-            options = {}
-        same_space = options.get("same_space", False)
+            my_options = {}
+        else:
+            my_options = options.copy()
+
+        # TODO #2498: access information for LFRic kernels do not have any
```

Comment:
The super validate would atm handle the situation that the variable name  could be different between the loops (obviously that won't be the case atm, but it could change), including renaming variables to be the same (if possible, so there are tests to make sure none of the two different loop variables is used in the other loop).
Long term (#257) they will do proper dependency testing, indeed based on the meta-data (well, based on the variable access info, which for now is taken from the meta-data).
BUT LFRic does have non-PSyIR nodes in the the ComponentIndices, which causes problems (#2498, and there is a #TODO for that).

Reply:
Since we recently agreed in the telco that #257 might just disallow stencil accesses to a variable written, this could indeed be checked for LFRic. So I've added a #todo for 257 to setting the force flag, so we can do it properly. Though this still needs to address the problem that LFRic VarAccesses has strings :(